docs/book/src/cronjob-tutorial/testdata/finalizer_example.go

```diff
@@ -67,10 +67,23 @@ func (r *CronJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		// then lets add the finalizer and update the object. This is equivalent
 		// to registering our finalizer.
 		if !controllerutil.ContainsFinalizer(cronJob, myFinalizerName) {
-			controllerutil.AddFinalizer(cronJob, myFinalizerName)
+			log.Info("Adding Finalizer for CronJob")
+			if ok := controllerutil.AddFinalizer(cronJob, myFinalizerName); !ok {
+				log.Error(err, "Failed to add finalizer into the custom resource")
+				return ctrl.Result{Requeue: true}, nil
+			}
+
 			if err := r.Update(ctx, cronJob); err != nil {
 				return ctrl.Result{}, err
 			}
+
+			// we re-fetch after having updated the CronJob, so that we have the latest
+			// state of the resource, and avoid the error "the object has been modified,
+			// please apply your changes to the latest version and try again".
+			if err := r.Get(ctx, req.NamespacedName, cronJob); err != nil {
+				log.Error(err, "unable to fetch CronJob")
+				return ctrl.Result{}, client.IgnoreNotFound(err)
+			}
```

Comment:
We need to change the code under hack/docs so that when we run `make generate-docs` we re-scaffold all and just add the right code on top to ensure that all is always updated.

Reply:
Indeed it will result in multi-version and crojob tutorial changed when you run `make generate-docs`